### PR TITLE
feat: Add New Redesign Of Issue Explorer

### LINKED ISSUE #507: [Advanced]: Add Issue Explorer Banner Image
### 🧑‍🔬 Advanced Issue

Welcome! This is an **[Advanced Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22advanced%22%20repo%3Ahiero-ledger%2Fhiero-website)** touching core website architecture.

It is designed for expert contributors who have demonstrated deep architectural understanding and a proven track record of high-quality contributions.

### 🐞 Problem Description

Currently there is a missing banner image for the Issue Explore Page. 

We would like to add one from our new redesign. 

### 🛠️ Implementation Notes

Implement the banner image defined in our figma file specification

https://www.figma.com/design/YndiSK4Z4P2yXPxlsj3UHU/Hiero-Website-Design-Darshit?node-id=0-1&p=f&t=CYNOHXDCWnteDqt6-0

PUSH WORK TO websiteRedesign BRANCH

Markdown Specification: 
https://github.com/hiero-ledger/hiero-website/blob/main/docs/design-specs.md

**Blog & Content Workflow:**
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md) — covers the current posting flow, front matter format, and content conventions

**Project Conventions:**
- Components live in `src/components/` — follow existing patterns for props, naming, and file structure
- Pages are under `src/app/` using the Next.js App Router
- Content (blog posts, etc.) lives in `content/`
- Styling uses [Tailwind CSS](https://tailwindcss.com/docs)

### 🧠 Advanced Contributors — Prerequisites & Expectations

> [!CAUTION]
> **Advanced issues are the highest-risk work in this project. We will reject PRs that do not meet these standards.**

### 🏁 Concrete Prerequisites
- **Advanced Web Development:** Proficient with React, Next.js, and TypeScript.
- **Expertise:** Deep architectural understanding of the website's component structure, routing, and build pipeline.
- **Proven History:** Successfully completed **≥ 10 intermediate issues** in this repo.
- **Consistency:** **≥ 3–4 months** of active, human-led contributions to this project.

> [!NOTE]
> **CI/CD Exception:** For issues focused on **GitHub Actions / Workflows**, the repo-specific thresholds above may be waived if the contributor demonstrates advanced-level proficiency in CI/CD.

### ⚠️ AI Usage Policy

- Using AI to generate code for Advanced issues is **strictly discouraged**
- AI may be used to help explain file relationships, but cannot be the main source of research.
- Submitting AI-generated or unvalidated code is grounds for **immediate closure**

### ⏱️ Timeline & Workflow
- **Typical time:** ~1 month / ~50 hours.
- 🔴 Completing an advanced issue in 1–3 days is a **red flag** and will likely be rejected.
- **Suggested:** Post your proposed architectural approach as a comment and wait for explicit maintainer approval **before writing any code.**

### 🔬 Technical Domains

- [ ] **Advanced React / Next.js** (Server Components, App Router, dynamic routing, middleware)
- [ ] **Component Architecture** (design systems, reusable patterns, composition)
- [ ] **Performance & Optimization** (Core Web Vitals, image optimization, code splitting, caching)
- [ ] **Accessibility** (WCAG compliance, screen reader support, keyboard navigation)
- [ ] **Build & Deployment** (Next.js config, CI/CD, hosting/deployment integration)
- [ ] **Testing** (unit, integration, E2E, accessibility testing, visual regression)

### 🛡️ Quality & Review Standards

Advanced PRs must be **"safe, maintainable, architecturally sound, and production-ready."**

1. **Architectural Fit:** The solution must fit naturally into the existing website architecture.
2. **Security & Correctness:** Evaluate all logic for injection risks, XSS vulnerabilities, or state corruption issues. 
3. **Maintainability:** Code must be short and clear enough for any other maintainer to debug without your assistance. 
4. **Backward Compatibility:** Existing pages, routes, and components must not break. If a breaking change is required, it must be explicitly managed.
5. **Comprehensive Testing:** Must include unit and integration tests covering all new logic paths, edge cases, and failure modes. AI-generated tests based on AI-generated code are grounds for immediate rejection.

### ✅ PR Quality Checklist

Before opening your PR, the contributor must confirm:
- [ ] I have spent the majority of my time researching the problem and solution space extensively, including reviewing relevant code, documentation, and external resources.
- [ ] I understand the system-wide impact of these changes on affected modules and performance.
- [ ] The design fits with the current website's architectural approaches.
- [ ] I have tested my changes extensively across browsers and screen sizes.
- [ ] Every line of code is personally understood and explainable.

### 📚 Resources & Support

**References:**
- [Next.js Documentation](https://nextjs.org/docs)
- [React Documentation](https://react.dev)
- [Tailwind CSS Documentation](https://tailwindcss.com/docs)
- [Vitest Documentation](https://vitest.dev)
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md)

**🆘 Stuck?**
- [Community Calls](https://zoom-lfx.platform.linuxfoundation.org/meetings/hiero?view=week)
- [Discord](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/discord.md)

### LINKED ISSUE #508: [Advanced]: Update Issue Explore Layout
### 🧑‍🔬 Advanced Issue

Welcome! This is an **[Advanced Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22advanced%22%20repo%3Ahiero-ledger%2Fhiero-website)** touching core website architecture.

It is designed for expert contributors who have demonstrated deep architectural understanding and a proven track record of high-quality contributions.

### 🐞 Problem Description

Currently the issue explorer layout is missing an eye catching UI

We would like to expand the design follow our outlined specification in figma

### 🛠️ Implementation Notes

Implement the design according to the figma file found here 

https://www.figma.com/design/YndiSK4Z4P2yXPxlsj3UHU/Hiero-Website-Design-Darshit?node-id=0-1&p=f&t=CYNOHXDCWnteDqt6-0

The design should include features that enhance a users accessibility of our site. 

The desgin should follow the specification outlined in our figma file.

Note:
For the skill labels, our team would like these positioned next to the repository labels. 

An example of this can be found here: https://www.figma.com/design/hZztcL7ukzpYLlIklBFzwP/Hiero-Website-Design-Daniel?node-id=0-1&p=f&t=3UYJ7D1bhd7NWafv-0

PUSH WORK TO websiteRedesign BRANCH

Markdown Specification: 
https://github.com/hiero-ledger/hiero-website/blob/main/docs/design-specs.md

**Blog & Content Workflow:**
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md) — covers the current posting flow, front matter format, and content conventions

**Project Conventions:**
- Components live in `src/components/` — follow existing patterns for props, naming, and file structure
- Pages are under `src/app/` using the Next.js App Router
- Content (blog posts, etc.) lives in `content/`
- Styling uses [Tailwind CSS](https://tailwindcss.com/docs)

### 🧠 Advanced Contributors — Prerequisites & Expectations

> [!CAUTION]
> **Advanced issues are the highest-risk work in this project. We will reject PRs that do not meet these standards.**

### 🏁 Concrete Prerequisites
- **Advanced Web Development:** Proficient with React, Next.js, and TypeScript.
- **Expertise:** Deep architectural understanding of the website's component structure, routing, and build pipeline.
- **Proven History:** Successfully completed **≥ 10 intermediate issues** in this repo.
- **Consistency:** **≥ 3–4 months** of active, human-led contributions to this project.

> [!NOTE]
> **CI/CD Exception:** For issues focused on **GitHub Actions / Workflows**, the repo-specific thresholds above may be waived if the contributor demonstrates advanced-level proficiency in CI/CD.

### ⚠️ AI Usage Policy

- Using AI to generate code for Advanced issues is **strictly discouraged**
- AI may be used to help explain file relationships, but cannot be the main source of research.
- Submitting AI-generated or unvalidated code is grounds for **immediate closure**

### ⏱️ Timeline & Workflow
- **Typical time:** ~1 month / ~50 hours.
- 🔴 Completing an advanced issue in 1–3 days is a **red flag** and will likely be rejected.
- **Suggested:** Post your proposed architectural approach as a comment and wait for explicit maintainer approval **before writing any code.**

### 🔬 Technical Domains

- [ ] **Advanced React / Next.js** (Server Components, App Router, dynamic routing, middleware)
- [ ] **Component Architecture** (design systems, reusable patterns, composition)
- [ ] **Performance & Optimization** (Core Web Vitals, image optimization, code splitting, caching)
- [ ] **Accessibility** (WCAG compliance, screen reader support, keyboard navigation)
- [ ] **Build & Deployment** (Next.js config, CI/CD, hosting/deployment integration)
- [ ] **Testing** (unit, integration, E2E, accessibility testing, visual regression)

### 🛡️ Quality & Review Standards

Advanced PRs must be **"safe, maintainable, architecturally sound, and production-ready."**

1. **Architectural Fit:** The solution must fit naturally into the existing website architecture.
2. **Security & Correctness:** Evaluate all logic for injection risks, XSS vulnerabilities, or state corruption issues. 
3. **Maintainability:** Code must be short and clear enough for any other maintainer to debug without your assistance. 
4. **Backward Compatibility:** Existing pages, routes, and components must not break. If a breaking change is required, it must be explicitly managed.
5. **Comprehensive Testing:** Must include unit and integration tests covering all new logic paths, edge cases, and failure modes. AI-generated tests based on AI-generated code are grounds for immediate rejection.

### ✅ PR Quality Checklist

Before opening your PR, the contributor must confirm:
- [ ] I have spent the majority of my time researching the problem and solution space extensively, including reviewing relevant code, documentation, and external resources.
- [ ] I understand the system-wide impact of these changes on affected modules and performance.
- [ ] The design fits with the current website's architectural approaches.
- [ ] I have tested my changes extensively across browsers and screen sizes.
- [ ] Every line of code is personally understood and explainable.

### 📚 Resources & Support

**References:**
- [Next.js Documentation](https://nextjs.org/docs)
- [React Documentation](https://react.dev)
- [Tailwind CSS Documentation](https://tailwindcss.com/docs)
- [Vitest Documentation](https://vitest.dev)
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md)

**🆘 Stuck?**
- [Community Calls](https://zoom-lfx.platform.linuxfoundation.org/meetings/hiero?view=week)
- [Discord](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/discord.md)

### PR DESCRIPTION
**Pull request by aceppaluni**
# Description

## Summary

Redesign the Issue Explorer page to better align with the Hiero website's visual design and design specifications.

## Changes

- Redesigned the Issue Explorer hero section with:
  - Hiero brand gradient background
  - Large "Issue Explorer" heading
  - Open-source / Hiero SDKs label
  - Open issue count and contribution badges
  - Decorative network-style background elements
- Redesigned the filter bar with:
  - Styled difficulty selector
  - Styled repository selector
  - Issue count indicator
- Redesigned issue results as responsive cards:
  - Repository badge
  - Issue title
  - Difficulty badge
  - Hover states and issue navigation arrow
- Added responsive layouts:
  - Single-column layout on mobile
  - Two-column layout on tablet
  - Four-column layout on large screens
- Applied the Hiero design specification colors and typography.
- Added visual styling for the four difficulty levels:
  - Good First Issue
  - Beginner
  - Intermediate
  - Advanced
- Preserved the existing issue filtering and GitHub issue-link behavior.

## Design

The updated layout is based on the provided Issue Explorer designs and follows the existing Hiero design system, including:

- Space Grotesk typography
- Hiero red palette
- Charcoal and neutral colors
- Existing type scale conventions
- Consistent borders, spacing, and card styling

## Testing

- Verified the Issue Explorer renders correctly with the updated layout.
- Verified difficulty and repository filters continue to function.
- Verified issue cards link to the corresponding GitHub issues.
- Verified the layout responds across different screen sizes.

## Related Issues

Fixes: #507
Fixes: #508 

## Checklist
- [ ] Tests added/updated
- [ ] Documentation updated
- [ ] Linting passes
- [x] Branch up-to-date with main

## Notes
This has been pushed to the main branch for now so we may view this in netlify. 

I will push this to the proper branch once approvals have been added. 

Please also note: that labels such as "Beginner, Good First Issue" may appear on one issue. This is because some SDK's for example C++ have added both labels to their issues. I have tested this by removing a label and it does correct this. I have left the other issues this way as I did not make the issues for other respective repositories. 

## Summary by CodeRabbit

- **New Features**
  - Added a redesigned Issue Explorer with a hero section, issue counts, filters, and responsive issue cards.
  - Issue cards now display repository names, titles, difficulty labels, and links to external issues.
  - Added an animated network-style background with node connections, grid lines, resizing, and reduced-motion support.
- **Bug Fixes**
  - Improved consistency of difficulty labels and their color classification.
  - Added clearer loading, error, and empty-result states.

### CODE REVIEW

File: src/app/issues/page.tsx [hunk 1]
```diff
@@ -21,63 +469,206 @@ export default function IssueExplorer() {
...
+              </p>
+
+              <div className="mt-8 flex flex-wrap gap-3">
+                <span className="rounded-full border border-white/20 bg-white/15 px-5 py-2.5 text-sm font-medium text-white backdrop-blur-sm">
+                  {issues.length} open issues
+                </span>
+
+                <span className="rounded-full border border-white/20 bg-white/15 px-5 py-2.5 text-sm font-medium text-white backdrop-blur-sm">
+                  Ready to contribute
+                </span>
```
Comment (danielmarv) [inline src/app/issues/page.tsx:531]: these should be buttons linking so some where so they are missing an <a></a> tag

**Review by danielmarv — CHANGES_REQUESTED** [review]
I like the implementatiom 
i think we should have pargination for the issues list

**Review by danielmarv — APPROVED** [review]
going to merge this for now
